services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/utils/FloodlightDashboardLogger.java

```diff
@@ -59,53 +63,58 @@ public void onPortEvent(DatapathId dpId, OFPortDesc portDesc, PortChangeType typ
      */
     public void onSwitchEvent(DatapathId dpId, SwitchChangeType state) {
         Map<String, String> context = makeSwitchContext(dpId);
-        context.put(FIELD_STATE, state.toString());
+        populateState(context, state.toString());
 
-        proceed(String.format("OF switch event (%s - %s)", dpId, state), context);
+        invokeLogger(String.format("OF switch event (%s - %s)", dpId, state), context);
     }
 
-    /**
-     * Build and write log message and MDC custom fields.
-     *  @param message a message text.
-     * @param logData a data for MDC custom fields.
-     */
-    protected void proceed(String message, Map<String, String> logData) {
-        Map<String, String> oldValues = MDC.getCopyOfContextMap();
-        logData.forEach(MDC::put);
-        try {
-            logger.info(message);
-        } finally {
-            for (String key : logData.keySet()) {
-                String original = oldValues.get(key);
-                if (original != null) {
-                    MDC.put(key, original);
-                } else {
-                    MDC.remove(key);
-                }
-            }
-        }
+    private void onPortUpDown(DatapathId dpId, OFPortDesc portDesc, String event) {
+        final OFPort ofPort = portDesc.getPortNo();
+        Map<String, String> context = makePortContext(dpId, ofPort);
+        populateState(context, event);
+
+        String switchPort = formatSwitchPort(dpId, ofPort);
+        invokeLogger(String.format("OF port event %s for %s: %s", event, switchPort, portDesc), context);
+    }
+
+    private void opPortStatusChange(DatapathId dpId, OFPortDesc portDesc, PortChangeType type) {
```

Comment:
This method is an almost exact copy of the method above. Is it possible to reuse the method above in this case?